src/main/java/games/strategy/engine/framework/GameDataUtils.java

```diff
@@ -38,23 +35,22 @@ public static GameData cloneGameData(final GameData data, final boolean copyDele
    */
   @SuppressWarnings("unchecked")
   public static <T> T translateIntoOtherGameData(final T object, final GameData translateInto) {
-    try {
-      ByteArrayOutputStream sink = new ByteArrayOutputStream(1024);
-      try (final GameObjectOutputStream out = new GameObjectOutputStream(sink)) {
-        out.writeObject(object);
-      }
-      final ByteArrayInputStream source = new ByteArrayInputStream(sink.toByteArray());
-      sink = null;
-      final GameObjectStreamFactory factory = new GameObjectStreamFactory(translateInto);
-      final ObjectInputStream in = factory.create(source);
-      try {
-        return (T) in.readObject();
-      } catch (final ClassNotFoundException ex) {
-        // should never happen
-        throw new RuntimeException(ex);
+    try (final ByteArrayOutputStream sink = new ByteArrayOutputStream(1024);
+        final GameObjectOutputStream out = new GameObjectOutputStream(sink)) {
+      out.writeObject(object);
+
+      try (final ByteArrayInputStream source = new ByteArrayInputStream(sink.toByteArray())) {
```

Comment:
Potential problem here...  Because the outer try-with-resources block has not yet ended, `out` may not have been flushed, in which case, all data may not have made it to `sink`.  Thus, `source` may not contain the complete stream for input.

The general pattern for this type of interaction is

```java
final ByteArrayOutputStream sink = // ...
try (final SomeFilterOutputStream out = new SomeFilterOutputStream(sink)) {
  // write to out
} // <-- ensure out is flushed here
try (final ByteArrayInputStream source = new ByteArrayInputStream(sink.toByteArray());
    final SomeFilterInputStream in = new SomeFilterInputStream(source)) {
  // read from in
}
```